atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java

```diff
@@ -849,43 +849,76 @@ public void truncateTable(final String tableName) {
+                                            + " It will appear in system_traces with UUID=" + CassandraKeyValueServices.convertCassandraByteBufferUUIDtoString(recv_trace));
+                                }
+                            } else {
+                                client.truncate(internalTableName(tableName));
                             }
-                        } else {
-                            client.truncate(internalTableName(tableName));
                         }
+                        CassandraKeyValueServices.waitForSchemaVersions(client, "(" + tablesToTruncate.size() + " tables in a call to truncateTables)");
+                        return null;
                     }
-                    CassandraKeyValueServices.waitForSchemaVersions(client, "(" + tableNames.size() + " tables in a call to truncateTables)");
-                    return null;
-                }
 
-                @Override
-                public String toString() {
-                    return "truncateTables(" + tableNames.size() + " tables)";
-                }
-            });
-        } catch (UnavailableException e) {
-            throw new PalantirRuntimeException("Creating tables requires all Cassandra nodes to be up and available.");
-        } catch (Exception e) {
-            throw Throwables.throwUncheckedException(e);
-        } finally {
-            schemaMutationLock.unlock();
+                    @Override
+                    public String toString() {
+                        return "truncateTables(" + tablesToTruncate.size() + " tables)";
+                    }
+                });
+            } catch (UnavailableException e) {
+                throw new PalantirRuntimeException("Creating tables requires all Cassandra nodes to be up and available.");
+            } catch (Exception e) {
+                throw Throwables.throwUncheckedException(e);
+            } finally {
+                schemaMutationLock.unlock();
+            }
         }
     }
 
+    private Set<String> filterOutTrulyEmptyTables(Set<String> tableNames) {
+        final Set<String> nonEmptyTables = Sets.newHashSet();
+        SliceRange slice = new SliceRange(ByteBuffer.wrap(PtBytes.EMPTY_BYTE_ARRAY), ByteBuffer.wrap(PtBytes.EMPTY_BYTE_ARRAY), false, Integer.MAX_VALUE);
+        final SlicePredicate predicate = new SlicePredicate();
+        predicate.setSlice_range(slice);
+
+        for (final String tableName : tableNames) {
+            int results = 1;
+            try {
+                results = clientPool.runWithPooledResource(new FunctionCheckedException<Client, Integer, Exception>() {
+                    @Override
+                    public Integer apply(Client client) throws Exception {
+                        List<KeySlice> range_slices = client.get_range_slices(new ColumnParent(internalTableName(tableName)), predicate, new KeyRange(1), deleteConsistency);
```

Comment:
what if we're truncating because we want to get rid of something that's all tombstones?